### PR TITLE
Issue 853

### LINKED ISSUE #853: `Optuna` wrapper in `etna.auto.optuna`
### 🚀 Feature Request

Create class `etna.auto.optuna.Optuna` which would handle logic around using optuna:
- method `tune` runs `study.optimize` via given `runner`

### Proposal

`Optuna`:

```python
class OptunaDirection(str, Enum):

    maximize = "maximize"
    minimize = "minimize"


class Optuna:
    """Class for encapsulate work with Optuna."""

    def __init__(
        self,
        direction: OptunaDirection,
        sampler: optuna.samplers.BaseSampler,
        study_name: Optinal[str] = None,
        storage: Optional[optuna.storages.BaseStorage] = None,
    ) -> None:

        self.sampler = sampler
        self.storage = storage
        self.study_name = study_name
        self.direction = OptunaDirection(direction)
        self.study = optuna.create_study(
            storage=self.storage,
            study_name=self.study_name,
            direction=self.direction,
            sampler=self.sampler,
            load_if_exists=True,
        )

    def tune(self,
        objective: Callable,
        n_trials: Optional[int] = None,
        timeout: Optional[int] = None,
        runner = LocalRunner(...),
        **kwargs
    ) -> optuna.samplers.BaseSampler:
        runner(self.study.optimize, objective, n_trials=n_trials, timeout=timeout, **kwargs)
        return self.study
```

### Test cases

- Create toy objective with default TPESampler - it should work as expected
    - Try LocalRunner
    - Try ParallelLocalRunner

### Additional context

You could use code from  `etna_utils.auto.optuna`

### PR DESCRIPTION
**Pull request by martins0n**
## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [ ] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

## Closing issues

closes #853

### CODE REVIEW

File: etna/auto/optuna/wrapper.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        storage:
+            storage to use
+        pruner:
+            optuna pruner
+        directions:
+            directions to optimize in case of multi-objective optimization
+        load_if_exists:
+            load study from storage if it exists or raise exception if it doesn't
+        """
+        self.sampler = sampler
```
Comment (Mr-Geekman) [inline etna/auto/optuna/wrapper.py:53]: Why do we have to save these parameters if it isn't a child of `BaseMixin`?

File: tests/test_auto/optuna/test_wrapper.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+from optuna.pruners import MedianPruner
+from optuna.samplers import GridSampler
+from optuna.samplers import TPESampler
+from optuna.storages import InMemoryStorage
+from optuna.study import StudyDirection
+
+from etna.auto.optuna import Optuna
+
+
+@pytest.fixture()
```
Comment (Mr-Geekman) [inline tests/test_auto/optuna/test_wrapper.py:11]: May be we should remove parenthesis.

File: etna/auto/optuna/wrapper.py (hunk 1 above)
Comment (martins0n) [reply etna/auto/optuna/wrapper.py:53]: I guess we can delete them for now.

File: tests/test_auto/optuna/test_wrapper.py (hunk 2 above)
Comment (martins0n) [reply tests/test_auto/optuna/test_wrapper.py:11]: We have the same `()` in multiple places. Maybe we should add linters for example. Or make nothing

**Review by Mr-Geekman — APPROVED** [review]

**Pull request merged**